cmd/cagent/main.go

```diff
@@ -97,27 +99,25 @@ func main() {
 	if serviceInstallUserPtr != nil && *serviceInstallUserPtr != "" ||
 		serviceInstallPtr != nil && *serviceInstallPtr {
 		if *outputFilePtr != "" {
-			fmt.Println("Output file(-o) flag can't be used together with service install(-s) flag")
-			os.Exit(1)
+			log.Fatalln("Output file(-o) flag can't be used together with service install(-s) flag")
 		}
 
 		if *serviceUninstallPtr {
-			fmt.Println("Service uninstall(-u) flag can't be used together with service install(-s) flag")
-			os.Exit(1)
+			log.Fatalln("Service uninstall(-u) flag can't be used together with service install(-s) flag")
 		}
 	}
 
 	cfg, err := cagent.HandleAllConfigSetup(*cfgPathPtr)
 	if err != nil {
-		log.Fatalf("Failed to handle cagent configuration: %s", err.Error())
+		log.WithError(err).Fatalln("Failed to handle Cagent configuration")
 	}
 
 	ca := cagent.New(cfg, *cfgPathPtr, version)
 
 	handleFlagPrintConfig(*printConfigPtr, cfg)
 
 	if ((serviceInstallPtr == nil) || ((serviceInstallPtr != nil) && (!*serviceInstallPtr))) &&
-		((serviceInstallUserPtr == nil) || ((serviceInstallUserPtr != nil) && (*serviceInstallUserPtr == ""))) &&
+		((serviceInstallUserPtr == nil) || ((serviceInstallUserPtr != nil) && len(*serviceInstallUserPtr) == 0)) &&
```

Comment:
https://github.com/cloudradar-monitoring/cagent/pull/117#discussion_r267868494

I think there is an advantage. I prefer to use `len()` everywhere. The problem, in short, that Go is statically typed, so once you change the type, you'll need to rewrite a lot of code, that could stay the same if written using `len()` initially..

Reply:
Well, I think this is okay for new code, but I don't see any point changing the already written checks. It's not such a critical problem :) 
I think we can leave it as it is in current PR, but in future avoid modifying checks just for the sake of modifying.

Reply:
Please let the person who opened a question resolve it.
Otherwise it's easily missed what's the explanation.

> but in future avoid modifying checks just for the sake of modifying.

I agree

>  The problem, in short, is that Go is statically typed, so once you change the type, you'll need to rewrite a lot of code

I don't think this is a problem, even if it can be a bit annoying. If the type changes, I think it's good to be forced to have a look at the implications and not mask this behind the use of `len()`